packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java

```diff
@@ -218,12 +219,13 @@ SurfaceTextureEntry getFlutterTexture() {
     return flutterTexture;
   }
 
-  public void takePicture(String filePath, @NonNull final Result result) {
-    final File file = new File(filePath);
-
-    if (file.exists()) {
-      result.error(
-          "fileExists", "File at path '" + filePath + "' already exists. Cannot overwrite.", null);
+  public void takePicture(@NonNull final Result result) {
+    final File outputDir = applicationContext.getCacheDir();
+    final File file;
+    try {
+      file = File.createTempFile("CAP", ".jpg", outputDir);
```

Comment:
Can we handle  [SecurityException](https://developer.android.com/reference/java/io/File#createTempFile(java.lang.String,%2520java.lang.String)) too ?
I see we have static file name for pictures. We can save one picture?

Reply:
This is not the case, the `createTempFile` method will create a unique file name. The first parameter ("CAP" in this case) is a prefix that will be used to generate the name (see [File.createTempFile](https://docs.oracle.com/javase/7/docs/api/java/io/File.html#createTempFile(java.lang.String,%20java.lang.String,%20java.io.File) for details).

The actual file will look something like "CAP3602253894598046604.jpg"

Reply:
Good catch regarding the `SecurityException`, I will add support to handle this as well.

Reply:
done

Reply:
Great